crates/agglayer-certificate-orchestrator/src/error.rs

```diff
@@ -34,8 +34,27 @@ pub enum CertificationError {
     Deserialize { source: bincode::Error },
     #[error("internal error: {0}")]
     InternalError(String),
+    #[error("prover failed")]
+    ProverFailed(String),
+    #[error("prover returned unspecified error")]
+    ProverReturnedUnspecifiedError,
+    #[error("prover execution failed: {source}")]
+    ProverExecutionFailed {
+        #[source]
+        source: ProofError,
+    },
     #[error("Storage error: {0}")]
     Storage(#[from] agglayer_storage::error::Error),
+    #[error("rollup contract address not found")]
+    RollupContractAddressNotFound {
+        #[source]
+        source: L1RpcError,
+    },
+    #[error("Unable to find aggchain vkey")]
+    UnableToFindAggchainVkey {
+        #[source]
+        source: L1RpcError,
+    },
```

Comment:
The `#[source]` attribute is not needed if the field is called `source` (a couple more occurrences above).

[Docs](https://crates.io/crates/thiserror)